code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -440,3 +440,45 @@
 		M.emote(pick("twitch","laugh","frown"))
 	..()
 	. = 1
+
+/datum/reagent/drug/pumpup
+	name = "Pump-Up"
+	description = "Take on the world!"
+	reagent_state = LIQUID
+	color = "#e38e44"
+	overdose_threshold = 10
+
+/datum/reagent/drug/pumpup/on_mob_metabolize(mob/living/L)
+	..()
+	ADD_TRAIT(L, TRAIT_STUNRESISTANCE, type)
+
+/datum/reagent/drug/pumpup/on_mob_end_metabolize(mob/living/L)
+	REMOVE_TRAIT(L, TRAIT_STUNRESISTANCE, type)
+	..()
+
+/datum/reagent/drug/pumpup/on_mob_life(mob/living/carbon/M)
+	var/high_message = pick("Go! Go! GO!", "You feel ready...", "You feel invincible...")
+	M.Jitter(5)
+
+	if(prob(5))
+		to_chat(M, "<span class='notice'>[high_message]</span>")
```

Comment:
can you not put the pick in here and save a var?

Reply:
Could do but this is more readable imo?

Reply:
save a var then

Reply:
its in the wrong scope either way

Reply:
hmmm?

Reply:
oh yes I see

Reply:
he picks the messages at the start of the proc, with only 5% of it actually being used.

Reply:
I don't really agree with needing to save a var here anyway 😄 readability > saving a line of code

Reply:
Could you at least only create high_message within the prob(5) if statement so that it isn't created/assigned a value if the 5% chance for it to be used didn't trigger in that cycle?

Reply:
STOP RESOLVING IF YOU DONT MAKE CHANGES REEEE

Reply:
they're both readable, one just saves the var.

Me put foot in sand on this one.

Reply:
😭 I'll change it okay

Reply:
Tbh I thought floyd was agreeing with me so I said it again and resolved, but I just realized the issue